#143 light executed script

asolovieff opened the pull request:
#143
Add LightExecutedScript that only contains the name, checksum and identifier.
DatamaintainDriver.listExecutedScripts() now return a sequence of LightExecutedScripts instead of the full ExecutedScript.
Modified MongoDriver, JdbcDriver and tests accordingly.
@driccio

asolovieff committed:
feat #143 Add LightExecutedScript

asolovieff committed:
feat #143 DatamaintainDriver contract modification

- listExecutedScripts() now returns a sequence of LightExecutedScript
- dependencies modified in consequence

asolovieff committed:
feat #143 Add LightExecutedScript json parsing

asolovieff committed:
feat #143 MongoDriver.listExecutedScripts() modification + test

asolovieff committed:
feat #143 JdbcDriver.listExecutedScripts() modification + test

Lysoun commented on modules/core/src/main/kotlin/datamaintain/core/script/Script.kt:
```diff
@@ -12,6 +12,36 @@ interface Script {
+        return "LightExecutedScript(name='$name', checksum='$checksum', identifier='$identifier')"
+    }
+}
+
```
Since it was discussed between the team, I would add a comment somewhere to explain why ```LightExecutedScript``` is not a data class and thus has to implement all those methods

asolovieff replied:
Not sure about it since it only comes from the fact that data class can't be inherited from and that ExecutedScript has to inherit from it.

Lysoun replied:
Ok

Lysoun commented on modules/driver-jdbc/src/test/kotlin/datamaintain/db/driver/jdbc/JdbcDriverTest.kt:
```diff
@@ -138,14 +141,16 @@ internal class JdbcDriverTest {
-                        script3,
-                        script2
+                        lightScript3,
+                        lightScript2
```
Since ```ExecutedScript``` extends ```LightExecutedScript```, shouldn't it work without even without giving explicit ```LightExecutedScript```? Did you try?

Lysoun replied:
If it does not, maybe add an extension to ExecutedScript in the tests that creates the light executed script associated to a script (basically by casting it I think)

asolovieff replied:
No it does not work without explicit type.

asolovieff replied:
And casting is not enough.

Lysoun replied:
Then you could add a function extension to ```ExecutedScript```, ```toLightExecutedScript()```, in the tests. I'm sure it will help for other tests because ```MongoDriverTest``` should also check stuff like that.

asolovieff replied:
Already done

Lysoun commented on modules/driver-mongo/src/main/kotlin/datamaintain/db/driver/mongo/serialization/KJsonParser.kt:
```diff
@@ -26,25 +48,25 @@ data class ExecutedScriptDb(@SerialName("_id") @ContextualSerialization val id:
+    executionStatus,
+    action,
+    executionDurationInMillis,
+    executionOutput
```
Is this mapping function still necessary? It appears to me that only light executed scripts will be read from the database from now on

asolovieff replied:
Indeed

Lysoun commented on modules/driver-mongo/src/main/kotlin/datamaintain/db/driver/mongo/serialization/KJsonParser.kt:
```diff
@@ -12,6 +14,26 @@ import kotlinx.serialization.json.Json
+    name = name,
+    checksum = checksum,
+    identifier = identifier
+)
```
Is this function necessary? Light executed scripts should not be inserted in the database

asolovieff replied:
Indeed

Lysoun commented on modules/driver-mongo/src/main/kotlin/datamaintain/db/driver/mongo/serialization/KJsonParser.kt:
```diff
@@ -61,4 +83,16 @@ class KJsonParser: ExecutedScriptJsonParser {
+    override fun serializeLightExecutedScript(lightExecutedScript: LightExecutedScript): String {
+        val lightExecutedScriptDb = lightExecutedScript.toLightExecutedScriptDb()
+
+        return mapper.stringify(LightExecutedScriptDb.serializer(), lightExecutedScriptDb)
```
Are light executed scripts serialized?

asolovieff replied:
No they are not.

Lysoun reviewed (changes requested):
I'm not so sure about reading the identifiers. They are used during the scripts execution to determine in what order the scripts should be played but are there relevant otherwise?
Since the objective here is to serialize ```ExecutedScript``` and only read ```LightExecutedScript```, I think the code to map executed scripts db to executed scripts could be removed and no code to serialize ```LightExecutedScript``` is necessary.

asolovieff committed:
refacto #143 Function toLightExecutedScript() in tests

Add an extension function toLightExecutedScript() inside test to avoid lightScript variable in addition to script variable.

asolovieff committed:
refacto #143 Removed useless methods in driver-mongo

ExecutedScript are only serialized and LightExecutedScript are only deserialized.

asolovieff committed:
refacto #143 Deleted a non used file in driver-jdbc

driccio reviewed (approved):
Hi, thanks for the PR.
